internal/clients/team4/client.go

```diff
@@ -2,16 +2,207 @@
 package team4
 
 import (
+	"fmt"
+	"math"
+	"reflect"
+	"sort"
+
 	"github.com/SOMAS2020/SOMAS2020/internal/common/baseclient"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
 	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
 )
 
 const id = shared.Team4
 
 func init() {
-	baseclient.RegisterClientFactory(id, func() baseclient.Client { return baseclient.NewClient(id) })
+	team4client := client{
+		BaseClient:    baseclient.NewClient(id),
+		clientJudge:   judge{BaseJudge: &baseclient.BaseJudge{}, t: nil},
+		clientSpeaker: speaker{BaseSpeaker: &baseclient.BaseSpeaker{}},
+		yes:           "",
+		obs:           &observation{},
+		internalParam: &internalParameters{},
+		savedHistory:  &map[uint]map[shared.ClientID]judgeHistoryInfo{},
+	}
+	team4client.clientJudge.parent = &team4client
+	team4client.clientSpeaker.parent = &team4client
+
```

Comment:
can you move this outside to a `NewClient` function? This will preserve your state between runs on the same instance, I'm afraid.

Reply:
see how team1 is doing it